pkg/controller/v1alpha1/trainedmodel/controller.go

```diff
@@ -190,6 +199,45 @@ func (r *TrainedModelReconciler) updateStatus(req ctrl.Request, desiredModel *v1
 	return nil
 }
 
+func (r *TrainedModelReconciler) updateConditions(req ctrl.Request, tm *v1alpha1api.TrainedModel) error {
+	// Get the parent inference service
+	isvc := &v1beta1api.InferenceService{}
+	if err := r.Get(context.TODO(), types.NamespacedName{Namespace: req.Namespace, Name: tm.Spec.InferenceService}, isvc); err != nil {
+		return err
+	}
+
+	var conditionErr error = nil
+	// Update Inference Service Ready condition
+	if isvc.Status.IsReady() {
+		log.Info("Parent InferenceService is ready", "TrainedModel", tm.Name, "InferenceService", isvc.Name)
+		tm.Status.SetCondition(v1alpha1api.InferenceServiceReady, &apis.Condition{
+			Status: v1.ConditionTrue,
+		})
+	} else {
+		log.Info("Parent InferenceService is not ready", "TrainedModel", tm.Name, "InferenceService", isvc.Name)
+		tm.Status.SetCondition(v1alpha1api.InferenceServiceReady, &apis.Condition{
+			Type:    v1alpha1api.InferenceServiceReady,
+			Status:  v1.ConditionFalse,
+			Reason:  "InferenceServiceNotReady",
+			Message: "Inference Service needs to be ready before Trained Model can be ready",
+		})
+
+		conditionErr = utils.FirstNonNilError([]error{
```

Comment:
why can't directly assign 
`conditionErr := fmt.Errorf(InferenceServiceNotReady, isvc.Name, tm.Name)` ?

Reply:
I wanted to make it FirstNonNilError because I could easily insert another condition check above it without having to worry about modifying that part. The reason why i declared it as a var is because `conditionErr` could potentially be used by other condition check.